app/actions/handlers.py

```diff
@@ -0,0 +1,38 @@
+import datetime
+import httpx
+import logging
+import stamina
+
+import app.services.gundi as gundi_tools
+import app.settings.integration as settings
+import app.actions.client as client
+
+from app.actions.configurations import AuthenticateConfig, FetchStudyIndividualsConfig, FetchIndividualEventsConfig
+from app.services.activity_logger import activity_logger
+from app.services.state import IntegrationStateManager
+
+
+logger = logging.getLogger(__name__)
+
+
+state_manager = IntegrationStateManager()
+
+
+async def action_auth(integration, action_config: AuthenticateConfig):
+    logger.info(
+        f"Executing auth action with integration {integration} and action_config {action_config}..."
+    )
+    mb_client = client.MovebankClient(
+        base_url=integration.base_url,
+        username=action_config.username,
+        password=action_config.password
+    )
+
+    token = await mb_client.get_token()
+
+    if token:
+        logger.info(f"Auth successful for integration '{integration.name}'. Token: '{token['api-token']}'")
+        return True
+    else:
+        logger.error(f"Auth unsuccessful for integration {integration}.")
+        return False
```

Comment:
To be consistent across handlers and integrations, until we define some convention, it'd be better to return a dict like this:
```suggestion
    if token:
        logger.info(f"Auth successful for integration '{integration.name}'. Token: '{token['api-token']}'")
        return {"valid_credentials": True}
    else:
        logger.error(f"Auth unsuccessful for integration {integration}.")
        return {"valid_credentials": False}
```

Reply:
Yes, And we need to return a dictionary either way for this case. The web-code will hang on any response that isn't a dictionary.

As @marianobrc said, we'll devise a standard for these responses soon, hopefully in the next sprint so we can begin retrofitting soon after.